pkg/pool-hooks/contracts/FeeTakingHookExample.sol

```diff
@@ -161,23 +177,34 @@ contract FeeTakingHookExample is BaseHooks, Ownable {
         return (true, hookAdjustedAmountsOutRaw);
     }
 
-    // Setters
+    // Permissioned functions
 
-    // Sets the hook swap fee percentage, which will be accrued after a swap was executed. This function must be
-    // permissioned.
+    /**
+     * @notice Sets the hook swap fee percentage, charged on every swap operation.
+     * @dev This function must be permissioned.
+     */
     function setHookSwapFeePercentage(uint64 feePercentage) external onlyOwner {
         hookSwapFeePercentage = feePercentage;
     }
 
-    // Sets the hook add liquidity fee percentage, which will be accrued after an add liquidity operation was executed.
-    // This function must be permissioned.
-    function setAddLiquidityHookFeePercentage(uint64 hookFeePercentage) public onlyOwner {
+    /**
+     * @notice Sets the hook add liquidity fee percentage, charged on every add liquidity operation.
+     * @dev This function must be permissioned.
+     */
+    function setAddLiquidityHookFeePercentage(uint64 hookFeePercentage) external onlyOwner {
         addLiquidityHookFeePercentage = hookFeePercentage;
     }
 
-    // Sets the hook remove liquidity fee percentage, which will be accrued after a remove liquidity operation was
-    // executed. This function must be permissioned.
-    function setRemoveLiquidityHookFeePercentage(uint64 hookFeePercentage) public onlyOwner {
+    /**
+     * @notice Sets the hook remove liquidity fee percentage, charged on every remove liquidity operation.
+     * @dev This function must be permissioned.
+     */
+    function setRemoveLiquidityHookFeePercentage(uint64 hookFeePercentage) external onlyOwner {
         removeLiquidityHookFeePercentage = hookFeePercentage;
     }
+
+    /// @notice Withdraws the accumulated fees. This function must be permissioned.
+    function withdrawFees(IERC20 feeToken) external onlyOwner {
```

Comment:
This doesn't strictly need to be permissioned if you're transferring the funds to the owner.

Reply:
Right; unless we had a recipient, which is overkill here.